CaesarCipher.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package caesarcipher;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ *
+ * @author kailabillie
+ */
+public class CaesarCipher {
+
+    /**
+     * @param args the command line arguments
+     */
+    public static void main(String[] args) {
+
+        String encodedString = Encode("-1:some text");
+        System.out.println(encodedString);
+
+    }
+
+    static String Encode(String word) {
+
+        String uppercase = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+        String lowercase = "abcdefghijklmnopqrstuvwxyz";
+        String digits = "1234567890";
+        
+        String [] words = word.split(":");
+        
+        
+        
+        int letterShift = Integer.parseInt(words[0]);
```

Comment:
Line 38 could be dangerous, since Strings could include characters that are not digits.
I would ask your interviewer some clarifying questions to decide which edge cases to check for. After, add some/all the checks  below line 34.
```
// if the word length == key length after splitting, input is missing the ':' symbol
if (word.length() == words[0].length()) throw new IllegalArgumentException("missing key");

// regex to remove all whitespace in the ciper key (could also just throw an exception if space found in loop below).
word[0].replaceAll("\\s+","");

// check for symbols or letters (exception made for '-' symbol only at the beginning of the string.
for (int i = 0; i < words[0].length(); i++) {
     if (Character.isLetter(words[0].charAt(i))) throw new IllegalArgumentException("key may not contain letters");
     if (!Character.isDigit(words[0].charAt(i))) && i != 0 && word.charAt(i) != '-') 
           throw new IllegalArgumentException("key may not contain symbols");
}
```